template/context.go

```diff
@@ -41,16 +49,21 @@ type Context struct {
 	}
 }
 
-// NewContextFromTask returns a new template context from the data available in
-// task. The provided context can then be used to populate runtime values in a
+// NewContext returns a new template context from the data available in the
+// task and the node where it is scheduled to run.
+// The provided context can then be used to populate runtime values in a
 // ContainerSpec.
-func NewContextFromTask(t *api.Task) (ctx Context) {
+func NewContext(n *api.NodeDescription, t *api.Task) (ctx Context) {
 	ctx.Service.ID = t.ServiceID
 	ctx.Service.Name = t.ServiceAnnotations.Name
 	ctx.Service.Labels = t.ServiceAnnotations.Labels
 
 	ctx.Node.ID = t.NodeID
-
+	ctx.Node.Hostname = n.Hostname
+	ctx.Node.Platform = Platform{
+		Architecture: n.Platform.Architecture,
+		OS:           n.Platform.OS,
+	}
```

Comment:
Let's make this tolerate a nil `n`, just in case `Describe` failed before storing the node description. In this case, we can just skip filling in the `ctx.Node` fields.